### PR TITLE
Config restructuring

### PR DESCRIPTION
**Pull request by kwitsch**
To declutter the global top level config options  i propose the grouping of ports and logging options as child options of top level options.

Example:
```YAML
ports:
  dns: 43
  http: 4000
  https: 4443
  tls: 853

log:
  level: warn
  format: json
  privacy: true
  timestamp: false
```

Full backward compatibility is provided through `validateConfig` and subsequently `fixDeprecatedLog` and `fixDeprecatedPorts`.

The documentation an unit tests are modified to use the new structure as default.

### CODE REVIEW

File: docs/config.yml [hunk 1]
```diff
@@ -234,14 +227,27 @@ hostsFile:
...
-# optional: obfuscate log output (replace all alphanumeric characters with *) for user sensitive data like request domains or responses to increase privacy. Default: false
-logPrivacy: false
+
+# optional: ports configuration
+ports:
+  # optional: DNS listener port(s) and bind ip address(es), default 53 (UDP and TCP). Example: 53, :53, "127.0.0.1:5353,[::1]:5353"
+  dns: 53
+  # optional: Port(s) and bind ip address(es) for DoT (DNS-over-TLS) listener. Example: 853, 127.0.0.1:853
+  tls: 853
+  # optional: HTTPS listener port(s) and bind ip address(es), default empty = no http listener. If > 0, will be used for prometheus metrics, pprof, REST API, DoH... Example: 443, :443, 127.0.0.1:443
```
Comment (0xERR0R) [inline docs/config.yml:237]: Description was outdated, I copied the text from the documentation page

```suggestion
  # optional: Port(s) and optional bind ip address(es) to serve HTTPS used for prometheus metrics, pprof, REST API, DoH... If you wish to specify a specific IP, you can do so such as 192.168.0.1:443. Example: 443, :443, 127.0.0.1:443,[::1]:443
```

File: docs/config.yml [hunk 2]
```diff
@@ -234,14 +227,27 @@ hostsFile:
...
+  https: 443
+  # http: 4000
```
Comment (0xERR0R) [inline docs/config.yml:239]:
```suggestion
  # optional: Port(s) and optional bind ip address(es) to serve HTTP used for prometheus metrics, pprof, REST API, DoH... If you wish to specify a specific IP, you can do so such as 192.168.0.1:4000. Example: 4000, :4000, 127.0.0.1:4000,[::1]:4000
  http: 4000 
```

**Review by 0xERR0R — COMMENTED** [review]
looks very good!

**Review by 0xERR0R — APPROVED** [review]

**Comment by kwitsch** [comment]
@0xERR0R ok i tried to resolve merge conflicts before merging but now the e2e test fails, did i do something wrong O.o

**Pull request merged**